### PR TITLE
Implement HF ViTForImageClassification models

### PR DESCRIPTION
**Pull request by aaron-siegel**
Adds a new UDF `pxtf.huggingface.vit_for_image_classification`.

### CODE REVIEW

**Comment by aaron-siegel** [comment]
This change is [Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/291)

File: pixeltable/functions/huggingface.py [hunk 1]
```diff
@@ -282,6 +282,80 @@ def detr_for_object_detection(image: Batch[PIL.Image.Image], *, model_id: str, t
...
+    {
+        'p': 0.494,  # class probability
+        'class': 935,  # class ID
+        'label': 'mashed potato',  # class label
+    }
+    ```
+
+    Examples:
+        Add a computed column that applies the model `google/vit-base-patch16-224` to an existing
+        Pixeltable column `tbl.image` of the table `tbl`:
```
Comment (mkornacker) [inline pixeltable/functions/huggingface.py]: column `image` of the ...

File: pixeltable/functions/huggingface.py [hunk 2]
```diff
@@ -282,6 +282,80 @@ def detr_for_object_detection(image: Batch[PIL.Image.Image], *, model_id: str, t
...
+    image: Batch[PIL.Image.Image],
+    *,
+    model_id: str,
+    top_k: int = 5
+) -> Batch[list[dict[str, Any]]]:
+    """
+    Computes image classifications for the specified image using a Vision Transformer (ViT) model.
+    `model_id` should be a reference to a pretrained [ViT Model](https://huggingface.co/docs/transformers/en/model_doc/vit).
+
+    __Note:__ Be sure the model is a ViT model that is trained for image classification, such as
```
Comment (mkornacker) [inline pixeltable/functions/huggingface.py]: should mention the specific class and include a link:
https://huggingface.co/docs/transformers/en/model_doc/vit#transformers.ViTForImageClassification

File: pixeltable/functions/huggingface.py [hunk 3]
```diff
@@ -282,6 +282,80 @@ def detr_for_object_detection(image: Batch[PIL.Image.Image], *, model_id: str, t
...
+        ... )
+    """
+    env.Env.get().require_package('transformers')
+    device = resolve_torch_device('auto')
+    import torch
+    from transformers import ViTImageProcessor, ViTForImageClassification
+
+    model: ViTForImageClassification = _lookup_model(model_id, ViTForImageClassification.from_pretrained, device=device)
+    processor = _lookup_processor(model_id, ViTImageProcessor.from_pretrained)
+    normalized_images = [normalize_image_mode(img) for img in image]
```
Comment (mkornacker) [inline pixeltable/functions/huggingface.py:339]: We need to get away from using this pattern (cpu-intensive operations that create shareable intermediate state) and instead switch to expr_udfs. Happy to discuss.

**Review by mkornacker — APPROVED** [review]
